mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/ChessFacade.kt

```diff
@@ -82,6 +86,48 @@ class ChessFacade(private val auth: Auth, private val store: Store) {
       it.filterNotNull().mapNotNull(ChessDocument::uid).map { uid -> StoreMatch(uid, store) }
     }
   }
+
+  /**
+   * Determines the [MatchResult] of a given [Match]
+   *
+   * @param match the [Match] to determine its [MatchResult]
+   */
+  suspend fun determineMatchOutcome(match: Match): MatchResult? {
+    val game = match.game.filterNotNull().first()
+    when (game.nextStep) {
+      is NextStep.Checkmate -> MatchResult.Reason.CHECKMATE
+      is NextStep.Stalemate -> Tie
+      else -> Tie
+    }
+    return Tie
+  }
+
+  /**
+   * Returns a [Flow] of a list of [ChessMatch] of the matches played by the given profile.
+   *
+   * @param profile the profile we want to know their played [ChessMatch]es
+   */
+  fun chessMatches(profile: Profile): Flow<List<ChessMatch>> {
+    val matches = matches(profile)
+    return matches.map { list -> list.mapNotNull { match -> match.toChessMatch(profile.uid) } }
+  }
+
+  /**
+   * Transforms a given [Match] to a [ChessMatch].
+   *
+   * @param currentUid the uid of the current user.
+   */
+  suspend fun Match.toChessMatch(
+      currentUid: String,
+  ): ChessMatch? {
+    val blackID = this.black.filterNotNull().first().uid
+    val whiteID = this.white.filterNotNull().first().uid
+    val adversary = if (blackID == currentUid) whiteID else blackID
+    val result = determineMatchOutcome(this)
+    val game = this.game.first()
+    val moveNum = game.serialize().size
+    return result?.let { ChessMatch(adversary, result, moveNum) }
+  }
```

Comment:
This should not be in the facade, and you should check with @barmettlerl how to make this conversion nicely and in an observable fashion.